gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -809,7 +812,7 @@ public GradeSaveResponse saveGrade(final Long assignmentId, final String student
 							log.debug("permissionGroupReference: " + permissionGroupReference);
 
 							// if permissions category is null (can grade all categories) or they match (can grade this category)
-							if (permissionCategoryId == null || permissionCategoryId.equals(gradeCategoryId)) {
+							if (!categoriesEnabled || (permissionCategoryId == null || permissionCategoryId.equals(gradeCategoryId))) {
```

Comment:
Similarly here, we only need to check the permission's category if categories are enabled.